json_output.go

```diff
@@ -0,0 +1,88 @@
+package golog
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"path/filepath"
+	"runtime"
+
+	"github.com/getlantern/hidden"
+)
+
+// JsonOutput creates an output that writes JSON structured log to different io.Writers for errors and debug
+func JsonOutput(errorWriter io.Writer, debugWriter io.Writer) Output {
+	return &jsonOutput{
+		E:  errorWriter,
+		D:  debugWriter,
+		pc: make([]uintptr, 10),
+	}
+}
+
+type jsonOutput struct {
+	// E is the error writer
+	E io.Writer
+	// D is the debug writer
+	D  io.Writer
+	pc []uintptr
+}
+
+type Event struct {
+	Message   string                 `json:"msg,omitempty"`
+	Component string                 `json:"component,omitempty"`
+	Caller    string                 `json:"caller,omitempty"`
+	Ops       map[string]interface{} `json:"ops,omitempty"`
```

Comment:
I feel like maybe there's a better name for this, but I'm not sure what it would be. Maybe `Context` or `Params` or something?

Reply:
I went with `Context`